pkg/storage/errors.go

```diff
@@ -96,15 +114,24 @@ func IsUnreachable(err error) bool {
 
 // IsTestFailed returns true if and only if err is a write conflict.
 func IsTestFailed(err error) bool {
-	return isErrCode(err, ErrCodeResourceVersionConflicts)
+	return isErrCode(err, ErrCodeResourceVersionConflicts, ErrCodeInvalidUID)
```

Comment:
To be clear, I agree we should differentiate ErrCodeInvalidUID and ErrCodeResourceVersionConflicts in the error returned by the storage package. My question is just an implementation detail :)